lib/services/profiles-api-service.js

```diff
@@ -81,6 +84,14 @@ function profileApiServiceFactory(
                         query.mac
                     );
                 }
+            }).then(function () {
+                if ( req.get(Constants.HttpHeaders.ApiProxyIp) ) {
+                    var proxy =  'http://%s:%s'.format(
+                        req.get(Constants.HttpHeaders.ApiProxyIp),
+                        req.get(Constants.HttpHeaders.ApiProxyPort)
+                    );
+                    return waterline.lookups.upsertProxyToMacAddress(proxy, query.mac);
+                }
```

Comment:
@shanemsully Do we need to return the lookup record if proxy IP header isn't set in the request?  It looks like the returned Promise will resolve to undefined if the header is missing.  Also, it looks like we may want to pass the node returned by the previous query into the .then callback and return immediately if it is undefined.